src/common/row_set.h

```diff
@@ -150,24 +150,31 @@ class PartitionBuilder {
     }
   }
 
+  // allocate thread local memory, should be called for each specific task
+  void AllocateForTask(size_t id) {
+    if (mem_blocks_[id].size() == 0) {
+      mem_blocks_[id].resize(1);
+    }
+  }
+
   common::Span<size_t> GetLeftBuffer(int nid, size_t begin, size_t end) {
     const size_t task_idx = GetTaskIdx(nid, begin);
-    return { mem_blocks_.at(task_idx).Left(), end - begin };
+    return { mem_blocks_.at(task_idx).data()->Left(), end - begin };
```

Comment:
Actually I'm confused by this, why do you need a vector if you are just using the first element?

Reply:
@trivialfis  Sorry for long response.
vector was changed to unique_ptr

Reply:
@trivialfis is it ok now?